### ISSUE TITLE
[Rule proposal] Enforce not.toHaveBeenCalled when expecting a function has not been called

### ISSUE THREAD

**Issue by pierrezimmermannbam** [jest-community/eslint-plugin-jest#1157]
For instance, the following cases would be invalid : 

```javascript
expect(mockFn).not.toHaveBeenCalledWith(params)

expect(mockFn).not.toHaveBeenCalledTimes(n)
```

This would be valid : 

```javascript
expect(mockFn).not.toHaveBeenCalled()
```

I think that when verifying that a function has not been called we should not expect that it hasn't been called with specific parameters and if the function has been called and we want to verify it hasn't been called with some other parameter we should verify that it has been called a certain number of times, each time with the expected parameter. 

This could help to avoid many false positive tests as accidentally using toHaveBeenCalledWith instead of toHaveBeenCalled would make the test pass if the function was called with any parameter other than undefined.

**Comment by MattAgn** [comment]
Totally agree!

**Comment by G-Rath** [comment]
This is what [`no-restricted-matchers`](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-restricted-matchers.md) is for:

```
{
  "jest/no-restricted-matchers": [
    "error",
    {
      "not.toHaveBeenCalledWith": null,
      "not.toHaveBeenCalledTimes": null
    }
  ]
}
```
